sites/all/modules/custom/language_selector_page/language_selector_page.module

```diff
@@ -142,3 +152,88 @@ function language_selector_page_preprocess_block(&$variables) {
     $variables['content'] = $content;
   }
 }
+
+/**
+ * Implements hook_url_outbound_alter().
+ *
+ * We alter the outbound url here, as this allows further modification and does
+ * not require JS. We simply check for the $_GET parameter and reappend it where
+ * it is needed (everywhere).
+ */
+function language_selector_page_url_outbound_alter(&$path, &$options, $original_path) {
+
+  // First we get our url parameters.
+  $parameters = drupal_get_query_parameters();
+
+  // First we check if we have a 2nd-language paramter, and the url we are
+  // processing is not external.
+  if (isset($parameters['2nd-language']) && !$options['external']) {
+
+    // Now lets check if we are not setting it already, because then we would
+    // override.
+    if (!isset($options['query']['2nd-language'])) {
+      // Lets not alter the autocomplete paths here.
+      if (strpos($path, 'autocomplete')) {
+        return;
+      }
+      else {
+        $options['query']['2nd-language'] = $parameters['2nd-language'];
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_drupal_goto_alter().
+ */
+function language_selector_page_drupal_goto_alter(&$path, &$options, &$http_response_code) {
+  // First we get our url parameters.
+  $parameters = drupal_get_query_parameters();
+
+  // Check if our parameter is set.
+  if (isset($parameters['2nd-language'])) {
+    $options['query']['2nd-language'] = $parameters['2nd-language'];
+  }
+}
+
+
+/**
+ * Implements hook_field_language_alter().
+ */
+function language_selector_page_field_language_alter(&$display_language, $context) {
+  // First we get our url parameters.
+  $parameters = drupal_get_query_parameters();
+
+  // Check if our parameter is set.
+  if (isset($parameters['2nd-language'])) {
+    locale_field_language_fallback($display_language, $context['entity'], $context['language']);
+  }
+}
+
+/**
+ * Implements hook_language_fallback_candidates_alter().
+ *
+ * Change the language fallback order.
+ */
+function language_selector_page_language_fallback_candidates_alter(array &$fallback_candidates) {
+  // First we get our url parameters.
+  $parameters = drupal_get_query_parameters();
+
+  // Add the default language as first fallback.
```

Comment:
This code should be on nexteuropa_multilingual feature, like on the commit d2968dc.
You can keep the second language in this module if you want, but the code for the default language must be on nexteuropa_multilingual feature.